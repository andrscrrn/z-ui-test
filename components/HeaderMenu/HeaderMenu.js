import React from "react";
import PropTypes from "prop-types";

import "./HeaderMenu.scss";

export default function HeaderMenu({ title }) {
  return (
    <div className="HeaderMenu">
      <h2 className="HeaderMenu__title">{title}</h2>
    </div>
  );
}

HeaderMenu.propTypes = {
  title: PropTypes.string.isRequired,
};
