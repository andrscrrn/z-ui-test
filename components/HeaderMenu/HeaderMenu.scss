@import "../../styles/responsive";
@import "../../styles/colors";

.HeaderMenu {
  $gray: rgba(0, 0, 0, .82);
  $light-gray: rgba(0, 0, 0, .53);
  background: linear-gradient(180deg, $gray 0%, $light-gray 49%, $transparent 100%);
  color: $white;
  display: flex;
  height: 150px;
  justify-content: center;
  align-items: center;
  position: absolute;
  width: 100%;

  &__container {
    align-items: center;
    display: flex;
    justify-content: space-between;
    max-width: $desktop-max-width;
    height: 40px;
    width: 100%;
  }

  &__title {
    font-size: 32px;
  }

  &__list {
    align-items: center;
    display: flex;
  }

  &__list-item {
    display: inline-block;
    margin-right: 50px;

    &:last-child {
      margin-right: 0;
    }
  }

  &__link {
    color: $white;
    font-size: 16px;
    font-weight: 300;
    letter-spacing: 0.2px;
  }
}
