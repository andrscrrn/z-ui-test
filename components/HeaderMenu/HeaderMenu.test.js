import React from "react";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import HeaderMenu from "./HeaderMenu";

Enzyme.configure({ adapter: new Adapter() });

describe("HeaderMenu", () => {
  it("renders correctly", () => {
    const props = {
      title: "",
      items: [],
    };
    const wrapper = shallow(<HeaderMenu {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
