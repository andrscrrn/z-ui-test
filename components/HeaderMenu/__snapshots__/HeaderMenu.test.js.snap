// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu renders correctly 1`] = `
<div
  className="HeaderMenu"
>
  <div
    className="HeaderMenu__container"
  >
    <h2
      className="HeaderMenu__title"
    />
    <ul
      className="HeaderMenu__list"
    >
      <li
        className="HeaderMenu__list-item"
        key="Log In / Sign Up"
      >
        <Link
          href="#"
        >
          <a
            className="HeaderMenu__link"
          >
            Log In / Sign Up
          </a>
        </Link>
      </li>
      <li
        className="HeaderMenu__list-item"
      >
        <Link
          href="#"
        >
          <a>
            <SearchIcon />
          </a>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
